content/dev-tools/tx-sender.page.tsx

```diff
@@ -0,0 +1,438 @@
+    const [dropsToSendForPayment, setDropsToSendForPayment] = useState(defaultDropsToSend)
+
+    // Escrow variables
+    const defaultFinishAfter = 60
+    const [finishAfter, setFinishAfter] = useState(defaultFinishAfter)
+    const [finishEscrowAutomatically, setFinishEscrowAutomatically] = useState(false)
+    const [escrowWidthPercent, setEscrowWidthPercent] = useState(0)
+
+    // Payment Channel variables
+    const defaultPaymentChannelAmount = 100000
+    const [paymentChannelAmount, setPaymentChannelAmount] = useState(defaultPaymentChannelAmount)
+
+    // Issued Currency / Trust Line Variables
+    const trustCurrencyCode = "FOO"
+
+    const defaultIssueAmount = 100
+    const [issueAmount, setIssueAmount] = useState(defaultIssueAmount)
+
+    const defaultTrustLimit = 100000
+    const [trustLimit, setTrustLimit] = useState(defaultTrustLimit)
+
+    const commonTxButtonParams = {
+        submitConstData,
+        connectionReady,
+        sendingWallet
+    }
+    
+    return (
+        <div className="row">
+            {/* TODO: Once xrpl.js 3.0 is released, replace this with a direct xrpl.js import */}
+            <script src="https://unpkg.com/xrpl@2.5.0-beta.0/build/xrpl-latest-min.js" async />
+            
+            <StatusSidebar balance={balance} sendingWallet={sendingWallet} connectionReady={connectionReady} txHistory={txHistory}/>
+            
+            <main className="main col-md-7 col-lg-6 order-md-3 page-tx-sender" role="main" id="main_content_body">
+                <section className="container-fluid pt-3 p-md-3">
+                    <h1>{translate("Transaction Sender")}</h1>
+                    <div className="content">
+                        <p>{translate("This tool sends transactions to the ")}
+                            <a href="dev-tools/xrp-faucets">{translate("XRP Testnet")}</a>
+                            {translate(" address of your choice so you can test how you monitor and respond to incoming transactions.")}
+                        </p>
+                        <form>
+                            <InitButton 
+                                isInitEnabled={isInitEnabled}
+                                toInit={{
+                                    existingClient: client,
+                                    alert,
+                                    setClient,
+                                    setBalance,
+                                    setSendingWallet,
+                                    setIsInitEnabled,
+                                    setConnectionReady,
+                                    partialPaymentParams
+                            }}/>
+
+                            <DestinationAddressInput 
+                                {...{defaultDestinationAddress, 
+                                destinationAddress, 
+                                setDestinationAddress, 
+                                isValidDestinationAddress, 
```

Comment:
I might be missing this but where is this used outside of DestinationAddressInput.  This validation could be encapsulate inside of the component and not need to be passed in.

Reply:
Good catch! :)